vcr/api/v2/registry.go

```diff
@@ -61,110 +61,94 @@ func (w *Wrapper) SearchVCs(ctx echo.Context) error {
-		switch c {
-		case credential.NutsOrganizationCredentialType:
-			return w.searchOrgs(ctx, untrusted, request.Query)
-		case credential.NutsAuthorizationCredentialType:
-			return w.searchAuths(ctx, untrusted, request.Query)
-		}
+	reader := jsonld.Reader{DocumentLoader: w.ContextManager.DocumentLoader()}
+	document, err := reader.Read(request.Query)
+	if err != nil {
+		return core.InvalidInputError("failed to convert query to JSON-LD expanded form: %w", err)
 	}
+	searchTerms := flatten(document, nil)
+	searchTerms = filter(searchTerms)
 
-	return core.InvalidInputError("given type not supported")
-}
-
-func (w *Wrapper) searchOrgs(ctx echo.Context, allowUntrusted bool, vcQuery vc.VerifiableCredential) error {
-	// credentialSubject struct that reflect current supported query params
-	type credentialSubject struct {
-		ID           string `json:"id"`
-		Organization struct {
-			Name string `json:"name"`
-			City string `json:"city"`
-		} `json:"organization"`
-	}
+	// sort terms to aid testing
+	sort.Slice(searchTerms, func(i, j int) bool {
+		left := strings.Join(searchTerms[i].IRIPath, "")
+		right := strings.Join(searchTerms[j].IRIPath, "")
 
-	var subjectsQuery []credentialSubject
-	if err := vcQuery.UnmarshalCredentialSubject(&subjectsQuery); err != nil {
-		return core.InvalidInputError("failed to unmarshall VC subject: %w", err)
-	}
+		return strings.Compare(left, right) < 0
+	})
 
-	// convert to query
-	query, err := w.VCR.Registry().QueryFor(concept.OrganizationConcept)
+	results, err := w.VCR.Search(ctx.Request().Context(), searchTerms, untrusted, nil)
 	if err != nil {
 		return err
 	}
-	if vcQuery.Issuer.String() != "" {
-		query.AddClause(concept.Eq("issuer", vcQuery.Issuer.String()))
-	}
-	if len(subjectsQuery) > 0 {
-		subjectQuery := subjectsQuery[0]
-		if subjectQuery.Organization.Name != "" {
-			query.AddClause(concept.Prefix("credentialSubject.organization.name", subjectQuery.Organization.Name))
-		}
-		if subjectQuery.Organization.City != "" {
-			query.AddClause(concept.Prefix("credentialSubject.organization.city", subjectQuery.Organization.City))
-		}
-		if subjectQuery.ID != "" {
-			query.AddClause(concept.Eq("credentialSubject.id", subjectQuery.ID))
+
+	return ctx.JSON(http.StatusOK, results)
+}
+
+// filter default required values that are empty in the query
+func filter(searchTerms []vcr.SearchTerm) []vcr.SearchTerm {
+	i := 0
+	for _, st := range searchTerms {
+		switch value := st.Value.(type) {
+		case string:
+			if value != "" && value != "0001-01-01T00:00:00Z" {
+				searchTerms[i] = st
+				i++
+			}
+		case float64:
+			if value != 0.0 {
+				searchTerms[i] = st
+				i++
+			}
 		}
+		// TODO bool problem, luckily there are no default bool fields in a VC
 	}
+	return searchTerms[:i]
+}
 
-	results, err := w.VCR.SearchLegacy(ctx.Request().Context(), query, allowUntrusted, nil)
-	if err != nil {
-		return err
+func flatten(document interface{}, currentPath []string) []vcr.SearchTerm {
```

Comment:
flattening is also a JSON-LD operation, so shouldn't it be part of the JSON-LD engine? (and transform it to VCR SearchTerms here)

Reply:
JSON-LD flatten is something different. The API needs to transform a JDON-LD document to a slice of `[]SearchTerm`, that logic is only used at this location.